explorer.js

```diff
@@ -358,27 +364,31 @@ function ViewController($scope, SharedService) {
 		});
 	});
 
-	$scope.renderObject = (data, _type, full) => {
-		// DEBUG.log("renderObject:", JSON.stringify(full));
-		const href = object2hrefvirt($scope.view.settings.bucket, data);
-
-		function render(d, href2, text, download) {
-			if (download) {
-				return `<a data-s3="object" data-s3key="${d}" href="${href2}" download="${download}">${text}</a>`;
-			}
-			return `<a data-s3="folder" data-s3key="${d}" href="${href2}">${text}</a>`;
+	function makeS3FileDownloadLink(etag, href, text, download) {
+		if (download) {
+			return `<a data-s3="object" data-etag="${etag}" href="${href}" download="${download}">${sanitizeString(text)}</a>`;
 		}
 
-		if (full.CommonPrefix) {
+		return `<a data-s3="folder" data-etag="${etag}" href="${href}">${sanitizeString(text)}</a>`;
+	}
+
+	$scope.renderObject = (s3FileKey, _type, s3File) => {
+		const href = object2hrefvirt($scope.view.settings.bucket, s3FileKey);
+
+		// ETag is in quotes so we need to remove them
+		const etag = s3File.ETag.split('"').join("");
```

Comment:
isn't a `replace` more straightforward than split and join ?

Reply:
> isn't a `replace` more straightforward than split and join ?

Yes this might be better, fixed